### DISCUSSION TITLE
Theme Colours (from Settings) should be retired (now that we have custom gradient coloring). Make the primary theme color dynamically pulled from Workspace gradient's dominant color.

### DISCUSSION THREAD

**Discussion by stripedgoat** [zen-browser/desktop#2371] [General]
Basically what I mentioned in the title.

If you pick for example Purple for your Theme Colour, and your workspace is colored with a different color (eg. Green), it's so ugly.

Theme Colours can be just made obsolete, and let the browser take the dominant colour of the actual workspace's coloring.

**Comment by jannisborgers** [comment]
Adding to this, the workspace color options are currently restricted by the fact that UI element and text is either dark or light **for all workspaces**, depending on the theme color.

If you want a workspace with a light background, and one with a dark background, choosing a light theme will only work on the light-colored workspace, but not on the dark-colored workspace, and vice versa. If you set the theme to auto, it might work until the OS triggers dark mode, then you can’t read the UI anymore 🙃

![zen-workspace-dark-light-ui-text-icons](https://github.com/user-attachments/assets/e9fc3ec5-62f8-499b-974e-d2b68a1dbd76)

## Possible solution: Add "UI color" selector
So instead of pulling the theme color from the workspace background color like OP suggests, I propose adding a "UI color" option in the workspace color picker, which changes the color of all UI elements in the toolbox (tab and button text, toolbar icons, URL bar text, etc.).

This way, workspaces can have any background color or gradient, and have a matching UI/foreground color, removing the conflict between workspace colors and theme colors.

I suspect this would override everything that Firefox themes do when installed, and also remove light/dark mode for the browser UI. I’m not sure if this is bad, though, given that the current situation can lead to the situation in my screenshots above. For total control over the colors, it would be nice to be able to make all these decisions for light and dark mode separately, so users can choose to have light workspaces muted in dark mode, or dark workspaces get boosted in light mode, for example.

**Reply by 12th-devs** [reply]
1000000% agree
